newapi start failed with panic

**Routine Checks**

[//]: # (Remove the space in the box and fill with an x)
+ [x] I have confirmed there are no similar issues currently
+ [x] I have confirmed I have upgraded to the latest version
+ [x] I have thoroughly read the project README, especially the FAQ section
+ [x] I understand and am willing to follow up on this issue, assist with testing and provide feedback 
+ [x] I understand and acknowledge the above, and understand that project maintainers have limited time and energy, **issues that do not follow the rules may be ignored or closed directly**

**Issue Description**
newapi start failed with panic
```txt
[SYS] 2025/10/29 - 17:13:46 | initializing token encoders
[SYS] 2025/10/29 - 17:13:46 | token encoders initialized
[SYS] 2025/10/29 - 17:13:46 | using MySQL as database
[SYS] 2025/10/29 - 17:13:46 | database migration started
[SYS] 2025/10/29 - 17:13:46 | system is already initialized at: 2025-10-14 10:38:12 +0800 CST
[SYS] 2025/10/29 - 17:13:46 | SYNC_FREQUENCY not set, use default value 60
[SYS] 2025/10/29 - 17:13:46 | Redis is enabled
[SYS] 2025/10/29 - 17:13:46 | New API v0.9.11 started
[SYS] 2025/10/29 - 17:13:46 | memory cache enabled
[SYS] 2025/10/29 - 17:13:46 | sync frequency: 60 seconds
[SYS] 2025/10/29 - 17:13:46 | InitChannelCache panic: assignment to entry in nil map, retrying once

2025/10/29 17:13:47 /build/model/ability.go:302 SLOW SQL >= 200ms
[252.696ms] [rows:0] TRUNCATE TABLE abilities
panic: assignment to entry in nil map [recovered, repanicked]

goroutine 1 [running]:
github.com/QuantumNous/new-api/model.InitChannelCache()
	/build/model/channel_cache.go:53 +0x945
github.com/QuantumNous/new-api/model.FixAbility()
	/build/model/ability.go:339 +0x879
main.main.func2.1()
	/build/main.go:77 +0x5b
panic({0x143e760?, 0x3be7c40?})
	/usr/local/go/src/runtime/panic.go:783 +0x132
github.com/QuantumNous/new-api/model.InitChannelCache()
	/build/model/channel_cache.go:53 +0x945
main.main.func2()
	/build/main.go:83 +0x30
main.main()
	/build/main.go:84 +0x19e
```

I added new remote api in Siliconflow channel, and it return 502 error, the server became broken.

I try to clean the channels table and it fixed. but I lost my data.

the data table is here.
+----+------+-----------------------------------------------------+----------------------+------------+--------+-------------+--------+--------------+------------+---------------+----------------------+-------+---------+----------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------+------------+---------------+---------------------+----------+----------+------------+------+---------------------------------------------------------------------------------------------------------------------------------------------------+----------------+-----------------+--------+---------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------+
| id | type | key                                                 | open_ai_organization | test_model | status | name        | weight | created_time | test_time  | response_time | base_url             | other | balance | balance_updated_time | models                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | group           | used_quota | model_mapping | status_code_mapping | priority | auto_ban | other_info | tag  | setting                                                                                                                                           | param_override | header_override | remark | channel_info                                                                                                                          | settings                                                                           |
+----+------+-----------------------------------------------------+----------------------+------------+--------+-------------+--------+--------------+------------+---------------+----------------------+-------+---------+----------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------+------------+---------------+---------------------+----------+----------+------------+------+---------------------------------------------------------------------------------------------------------------------------------------------------+----------------+-----------------+--------+---------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------+
|  1 |   40 | sk-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx | NULL                 |            |      1 | siliconflow |      0 |   1760409544 | 1761728792 |          2440 |                      |       |       0 |                    0 | moonshotai/Kimi-K2-Instruct-0905,Qwen/Qwen3-32B                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | default,Default |       2325 |               |                     |        0 |        1 |            |      | {"force_format":false,"thinking_to_content":false,"proxy":"","pass_through_body_enabled":false,"system_prompt":"","system_prompt_override":false} | NULL           | NULL            | NULL   | {"is_multi_key": false, "multi_key_mode": "random", "multi_key_size": 0, "multi_key_status_list": null, "multi_key_polling_index": 0} | {}                                                                                 |
|  3 |    1 | sk-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx |                      |            |      1 | ??AI        |      0 |   1761235951 | 1761728796 |          3617 | https://aicanapi.com |       |       0 |                    0 | claude-sonnet-4-5-20250929,gemini-2.5-pro,gemini-2.5-flash,gpt-5-2025-08-07,gpt-5-mini-2025-08-07,gpt-5-nano-2025-08-07,claude-haiku-4-5-20251001,claude-haiku-4-5-20251001-thinking,claude-sonnet-4-5-20250929-thinking,gemini-2.5-pro-exp,gemini-2.5-flash-preview-05-20,gemini-2.5-flash-image-preview,grok-4,chatgpt-4o-latest,flux-kontext-max,flux-kontext-pro,flux.1-kontext-dev,flux.1-kontext-pro,gpt-3.5-turbo,gpt-3.5-turbo-0125,gpt-3.5-turbo-0613,gpt-3.5-turbo-16k,gpt-3.5-turbo-16k-0613,gpt-4,gpt-4-0125-preview,gpt-4-0613,gpt-4-1106-preview,gpt-4-32k,gpt-4-32k-0613,gpt-4-turbo,gpt-4-turbo-2024-04-09,gpt-4-turbo-preview,gpt-4-vision-preview,gpt-4.1,gpt-4o,gpt-4o-2024-05-13,gpt-4o-2024-08-06,gpt-4o-2024-11-20,gpt-4o-all,gpt-4o-image,gpt-4o-image-vip,gpt-4o-mini,gpt-4o-mini-2024-07-18,gpt-5,gpt-5-chat,gpt-5-chat-2025-08-07,gpt-5-chat-latest,gpt-5-mini,gpt-5-nano,gpt-image-1,gpt-oss-120b,gpt-oss-20b,o1,o1-2024-12-17,o1-mini,o1-mini-2024-09-12,o1-preview,o1-preview-2024-09-12,o3-2025-04-16,o3-mini,o3-mini-2025-01-31,o3-mini-2025-01-31-high,o3-mini-2025-01-31-low,o3-mini-2025-01-31-medium,o3-mini-high,o3-mini-low,o3-mini-medium,text-embedding-3-large,text-embedding-3-small,text-embedding-ada-002,claude-3-5-sonnet-20240620,claude-3-7-sonnet-20250219,claude-3-7-sonnet-20250219-thinking,claude-3-7-sonnet-latest,claude-3-7-sonnet-thinking,claude-3.5-sonnet,claude-4-sonnet,claude-4-sonnet-thinking,claude-sonnet-4-20250514,claude-sonnet-4-20250514-t,claude-sonnet-4-20250514-thinking,grok-2-1212,grok-2-vision-1212,grok-3,grok-3-beta,grok-3-deepsearch,grok-3-fast-beta,grok-3-reasoning,grok-beta,gemini-2.0-flash-lite,gemini-2.0-flash-lite-001,gemini-2.5-flash-lite-preview-06-17 | default         |          0 |               |                     |        0 |        1 |            |      | {"force_format":false,"thinking_to_content":false,"proxy":"","pass_through_body_enabled":false,"system_prompt":"","system_prompt_override":false} | NULL           | NULL            | NULL   | {"is_multi_key": false, "multi_key_mode": "random", "multi_key_size": 0, "multi_key_status_list": null, "multi_key_polling_index": 0} | {"allow_service_tier":false,"disable_store":false,"allow_safety_identifier":false} |
+----+------+-----------------------------------------------------+----------------------+------------+--------+-------------+--------+--------------+------------+---------------+----------------------+-------+---------+----------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------+------------+---------------+---------------------+----------+----------+------------+------+---------------------------------------------------------------------------------------------------------------------------------------------------+----------------+-----------------+--------+---------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------+
**Steps to Reproduce**

**Expected Result**

**Related Screenshots**
If none, please delete this section.

--- a/model/channel_cache.go
+++ b/model/channel_cache.go
@@ -167,9 +167,19 @@ func GetRandomSatisfiedChannel(group string, model string, retry int) (*Channel,
 	// Calculate the total weight of all channels up to endIdx
 	totalWeight := sumWeight * smoothingFactor
 
-	// totalWeight 小于等于0时，设置为1，选择第一个渠道
+	// totalWeight 小于等于0时，给每个渠道加100的权重，然后进行随机选择
 	if totalWeight <= 0 {
-		totalWeight = 1
+		if len(targetChannels) > 0 {
+			totalWeight = len(targetChannels) * 100
+			randomWeight := rand.Intn(totalWeight)
+			for _, channel := range targetChannels {
+				randomWeight -= 100
+				if randomWeight <= 0 {
+					return channel, nil
+				}
+			}
+		}
+		return nil, errors.New("no available channels")
 	}
 	// Generate a random value in the range [0, totalWeight)
 	randomWeight := rand.Intn(totalWeight)

